src/FormModel.php

```diff
@@ -176,20 +176,22 @@ public function load(array|object|null $data, ?string $formName = null): bool
 
     public function setAttribute(string $name, mixed $value): void
     {
-        [$realName] = $this->getNestedAttribute($name);
-
-        if (isset($this->attributes[$realName])) {
-            /** @var mixed */
-            $value = match ($this->attributes[$realName]) {
-                'bool' => (bool) $value,
-                'float' => (float) $value,
-                'int' => (int) $value,
-                'string' => (string) $value,
-                default => $value,
-            };
-
-            $this->writeProperty($name, $value);
+        if ($value !== null) {
+            [$realName] = $this->getNestedAttribute($name);
+
+            if (isset($this->attributes[$realName])) {
+                /** @var mixed */
+                $value = match ($this->attributes[$realName]) {
+                    'bool' => (bool) $value,
+                    'float' => (float) $value,
+                    'int' => (int) $value,
+                    'string' => (string) $value,
+                    default => $value,
+                };
+            }
         }
+
+        $this->writeProperty($name, $value);
```

Comment:
After this change can be try write to non-exist property. Maybe add check on null into `if (isset($this->attributes[$realName])) {` ?

Reply:
done.